device_doctor/lib/src/android_device.dart

```diff
@@ -151,6 +152,23 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
       await device.recover();
     }
   }
+
+  @visibleForTesting
+  Future<HealthCheckResult> adbPowerServiceCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      final String result =
+          await eval('adb', <String>['shell', 'dumpsys', '-l', '|', 'grep', 'power\$'], processManager: processManager);
+      if (result.trim() == 'power') {
```

Comment:
Why we need to check for a string, is the exit code not enough in this situation?

Reply:
I am not sure if there are cases where other services are available, whereas the `power` service is not up. For this case, it will return 0, but with null value.

Reply:
Updated to run `adb shell dympsys power` directly. This way, it relies on exit code to determine success or failure.

Reply:
Awesome, thanks!